### PR TITLE
Feature/key list

### PR DESCRIPTION
**Pull request by ghostbuster91**
Please look at AppController and tell me what you think. Keep in mind that with a little help of DI we could easily test it. 
Also look at register_screen.xml

### CODE REVIEW

File: key-list-core/src/test/java/com/safechat/key/list/KeyListControllerTest.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        controller.start()
+
+        verify(view).showLoader()
+    }
+
+    @Test
+    fun shouldHideLoaderAfterCallForListOfKeysComplete() {
+        controller.start()
+
+        verify(view).showLoader()
```
Comment (OrdonTeam) [inline key-list-core/src/test/java/com/safechat/key/list/KeyListControllerTest.kt]: hide??

File: registration-core/src/main/java/com/safechat/register/RegisterController.kt [hunk 2]
```diff
@@ -16,6 +17,7 @@ class RegisterController(
                     .subscribe({ registerView.successLogIn() }, { registerView.showKeyRegisterError() })
```
Comment (OrdonTeam) [inline registration-core/src/main/java/com/safechat/register/RegisterController.kt]: missing  `onRegistrationCompletedListener.onRegistrationCompleted()`

**Comment by OrdonTeam** [comment]
It is definitely to early for such component. It makes code overengineered and so complicated that even you don't get it fully :(

**Comment by ghostbuster91** [comment]
I would rather said that it was late and I was tired, but got your point. What would you change to simplify it?

**Comment by ghostbuster91** [comment]
I see only one other way to do it which is to invoke transition to the next screen from `RegistrationViewImpl` by using some `Navigation` interface in order to avoid creation of dependencies between relevant compontents.

**Pull request closed without merging**